h/schemas/annotation.py

```diff
@@ -215,9 +218,9 @@ def validate(self, data):
             )
 
         if "target" in appstruct:
-            new_appstruct["target_selectors"] = _target_selectors(
-                appstruct.pop("target")
-            )
+            targets = appstruct.pop("target")
+            new_appstruct["target_description"] = _target_description(targets)
+            new_appstruct["target_selectors"] = _target_selectors(targets)
```

Comment:
Same thing but for the annotation update API (rather than create). Could deduplicate the code a bit better here.